cloud/services/networkinterfaces/networkinterfaces.go

```diff
@@ -143,14 +151,59 @@ func (s *Service) Delete(ctx context.Context, spec interface{}) error {
 	}
 	klog.V(2).Infof("deleting nic %s", nicSpec.Name)
 	err := s.Client.Delete(ctx, s.Scope.ResourceGroup(), nicSpec.Name)
-	if err != nil && azure.ResourceNotFound(err) {
-		// already deleted
-		return nil
-	}
-	if err != nil {
+	if err != nil && !azure.ResourceNotFound(err) {
 		return errors.Wrapf(err, "failed to delete network interface %s in resource group %s", nicSpec.Name, s.Scope.ResourceGroup())
 	}
-
-	klog.V(2).Infof("successfully deleted nic %s", nicSpec.Name)
+	NATRuleName := s.MachineScope.Name()
+	err = s.InboundNATRulesClient.Delete(ctx, s.Scope.ResourceGroup(), nicSpec.PublicLoadBalancerName, NATRuleName)
+	if err != nil && !azure.ResourceNotFound(err) {
```

Comment:
Seems like `err != nil && !azure.ResourceNotFound(err)` will be a very common delete case. Perhaps, in a future PR it might be nice to intro a util func for that.

Reply:
@devigned -- Would you mind capturing that in an issue, so we don't lose it?